dspace-api/src/test/java/org/dspace/content/dao/RelationshipDAOImplTest.java

```diff
@@ -0,0 +1,186 @@
+import org.dspace.content.EntityType;
+import org.dspace.content.Item;
+import org.dspace.content.Relationship;
+import org.dspace.content.RelationshipType;
+import org.dspace.content.WorkspaceItem;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.CommunityService;
+import org.dspace.content.service.EntityTypeService;
+import org.dspace.content.service.InstallItemService;
+import org.dspace.content.service.ItemService;
+import org.dspace.content.service.RelationshipService;
+import org.dspace.content.service.RelationshipTypeService;
+import org.dspace.content.service.WorkspaceItemService;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Created by: Andrew Wood
+ * Date: 20 Sep 2019
+ */
+public class RelationshipDAOImplTest extends AbstractIntegrationTest {
+
+    private static final Logger log = org.apache.logging.log4j.LogManager.getLogger(RelationshipDAOImplTest.class);
+
+    private Relationship relationship;
+
+    private Item itemOne;
+
+    private Item itemTwo;
+
+    private Collection collection;
+
+    private Community owningCommunity;
+
+    private RelationshipType relationshipType;
+
+    private List<Relationship> relationshipsList = new ArrayList<>();
+
+    private EntityType entityTypeOne;
+
+    private EntityType entityTypeTwo;
+
+    protected CommunityService communityService = ContentServiceFactory.getInstance().getCommunityService();
+    protected CollectionService collectionService = ContentServiceFactory.getInstance().getCollectionService();
+    protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    protected InstallItemService installItemService = ContentServiceFactory.getInstance().getInstallItemService();
+    protected WorkspaceItemService workspaceItemService = ContentServiceFactory.getInstance().getWorkspaceItemService();
+    protected RelationshipTypeService relationshipTypeService =
+            ContentServiceFactory.getInstance().getRelationshipTypeService();
+    protected RelationshipService relationshipService = ContentServiceFactory.getInstance().getRelationshipService();
+    protected EntityTypeService entityTypeService = ContentServiceFactory.getInstance().getEntityTypeService();
+
+    /**
+     * Initalize DSpace objects used for testing for each test
+     */
+    @Before
+    @Override
+    public void init() {
+        super.init();
+        try {
+            // Create objects for testing
+            context.turnOffAuthorisationSystem();
+            owningCommunity = communityService.create(null, context);
+            collection = collectionService.create(context, owningCommunity);
+            WorkspaceItem workspaceItem = workspaceItemService.create(context, collection, false);
+            WorkspaceItem workspaceItemTwo = workspaceItemService.create(context, collection, false);
```

Comment:
We should be using Builder classes to create any new test content, as those classes will automatically clean up after themselves.  See https://github.com/DSpace/DSpace/tree/master/dspace-server-webapp/src/test/java/org/dspace/app/rest/builder

Reply:
As noted below, I overlooked this is in the `dspace-api` layer which doesn't support Builders.  Marking as resolved. This is good enough as-is.